lib/modulr/client.rb

```diff
@@ -91,28 +93,22 @@ def auth_options(options)
     end
 
     def handle_request_error(error)
+      response = error.response
       case error
       when Faraday::ClientError
-        if error.response
-          handle_error_response(error)
+        case response[:status]
+        when 403
+          raise ForbiddenError, response
+        when 404
+          raise NotFoundError, response
         else
-          handle_network_error(error)
+          raise RequestError, response
         end
       else
-        raise error
+        raise Error, response
```

Comment:
We are not directly raising errors anymore. We are wrapping in domain errors that have a small accessor to retrieve the specific details in needed.

Reply:
We are currently capturing Faraday errors in our code: https://github.com/devengoapp/core/blob/602bc387b11d1d4ccb13fb44f9ecc6b14710dd51/app/domains/banking/services/transfer_processor.rb#L32

We should improve it because it's a kind of workaround for PFS timeouts, but just to be aware.

Reply:
Correct @IvanGuardado.

In the first stage, we should update this catch to reflect the new errors. 
 
Then the new provider-specific errors should be wrapped at the processor level in generic business errors so we don't have to deal with provider-specific errors here.